### PR TITLE
deployer: update logs APIs  to fetch the logs from shuttle-logger

### PR DESCRIPTION
**Pull request by iulianbarbu**
## Description of change

Updated `get_logs` and `get_logs_stream` deployer methods to be based on `shuttle-logger`.
This PR is rebased on top of #1139 .

## How has this been tested? (if applicable)

N/A, pending on #1139 .

### CODE REVIEW

File: deployer/src/handlers/mod.rs [hunk 1]
```diff
@@ -554,16 +555,24 @@ pub async fn start_deployment(
...
-    if let Some(deployment) = persistence.get_deployment(&deployment_id).await? {
+    let mut logs_request: tonic::Request<LogsRequest> = tonic::Request::new(LogsRequest {
+        deployment_id: deployment_id.to_string(),
+    });
+
+    logs_request.extensions_mut().insert(claim);
+
+    let mut client = deployment_manager.logs_fetcher().clone();
+    if let Ok(logs) = client.get_logs(logs_request).await {
+        // TODO: awaits on the From impl for `shuttle_proto::logger::LogItem` -> `shuttle_common::LogItem`.
```
Comment (oddgrd) [inline deployer/src/handlers/mod.rs:570]: @iulianbarbu this todo is done, right?

Comment (iulianbarbu) [reply deployer/src/handlers/mod.rs:570]: Yup, in theory. We should test it though, it's the main subject of this PR.

File: common/src/backends/tracing.rs [hunk 2]
```diff
@@ -34,12 +33,15 @@ where
         .unwrap();
     let fmt_layer = fmt::layer();
 
+    // The OTLP_ADDRESS env var is useful for setting a localhost address when running deployer locally.
+    let otlp_address = std::env::var("OTLP_ADDRESS").unwrap_or(OTLP_ADDRESS.into());
```
Comment (chesedo) [inline common/src/backends/tracing.rs:37]: Nice :partying_face:

File: deployer/src/handlers/mod.rs (hunk 1 above)
Comment (chesedo) [reply deployer/src/handlers/mod.rs:570]: This PR seems to update deployer correctly. So we can merge this and make another tickets if something is not working.

File: deployer/src/main.rs [hunk 3]
```diff
@@ -35,11 +40,24 @@ async fn main() {
         Some(args.auth_uri.to_string()),
     );
 
+    let channel = args
+        .logger_uri
+        .connect()
+        .await
+        .expect("failed to connect to resource recorder");
```
Comment (oddgrd) [inline deployer/src/main.rs:47]: Woops!
```suggestion
        .expect("failed to connect to logger");
```

File: deployer/src/handlers/mod.rs (hunk 1 above)
Comment (oddgrd) [reply deployer/src/handlers/mod.rs:570]: I agree.

**Pull request merged**